### PR TITLE
Lexical Grammar - ID_Start not applied to _ or $

### LINKED ISSUE #36610: Lexical Grammar - ID_Start not applied to _ or $
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers

### What specific section or headline is this issue about?

#identifiers

### What information was incorrect, unhelpful, or incomplete?

This is not an issue with the page so much as an issue in the page it links to, or the Unicode chars themselves.  This was the only place I could find to call it out.  This section of this page links [here](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5Cp%7BID_Start%7D). That page on util.unicode.org filters unicode chars using `\p{ID_Start}`, but it does not include `_` or `$` in the results.

Is there another place to report this?  Is it not a bug and something that the MDN docs page should mention?  Is it a non-issue?

### What did you expect to see?

I expect to see `_` or `$` in the search results.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by sidewayss**
Fixes #36610.

Now this paragraph aligns with the regular expression in the blue Notes block below it.

### CODE REVIEW

**Comment by sidewayss** [comment]
Wrong branch. Gotta start again.

File: files/en-us/web/javascript/reference/lexical_grammar/index.md [hunk 1]
```diff
@@ -158,7 +158,10 @@ class C {
...
+In JavaScript, identifiers are commonly made of alphanumeric characters, underscores (`_`), and dollar signs (`$`). Identifiers are not allowed to start with numbers. However, JavaScript identifiers are not only limited to {{Glossary("ASCII")}} — many Unicode code points are allowed as well. Namely:
+
+- Start characters can be any character in the [ID_Start](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5Cp%7BID_Start%7D) category plus `_` and `$`.
+- After the first character, you can use any character in the [ID_Continue](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5Cp%7BID_Continue%7D) plus U+200C (ZWNJ) and U+200D (ZWJ).
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/lexical_grammar/index.md:164]:
```suggestion
- After the first character, you can use any character in the [ID_Continue](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5Cp%7BID_Continue%7D) category plus U+200C (ZWNJ) and U+200D (ZWJ).
```

**Review by Josh-Cena — APPROVED** [review]
Great change, thank you!

File: files/en-us/web/javascript/reference/lexical_grammar/index.md (hunk 1 above)
Comment (sidewayss) [reply files/en-us/web/javascript/reference/lexical_grammar/index.md:164]: I made the same edit in a separate commit!

**Pull request closed without merging**